pennylane/transforms/defer_measurements.py

```diff
@@ -56,6 +56,16 @@ def _check_tape_validity(tape: QuantumTape):
                 "measurements on a device that does not support them."
```

Comment:
Somewhat aside, when I return `qml.expval(qml.PauliX(0))`, say, when I have postselection and all of my postselection is unsuccessful, I correctly get `nan`:

```
dev = qml.device("default.qubit", shots=10)

@qml.qnode(dev)
def f(x):
    for i in range(100):
        qml.RY(x, 0)
        m = qml.measure(0, postselect=1)
    return qml.expval(qml.PauliX(0))

x = np.array([0.4, 0.6])
f(x)
```

```
array([nan, nan])
```

Are we purposefully not showing the `UserWarning` after the first time it happens?

```
UserWarning: All shots were thrown away as invalid. This can happen for example when post-selecting the 1-branch of a 0-state. Make sure your circuit has some probability of producing a valid shot.
  warnings.warn(
```

Just wondering is all 😁

Reply:
That's just python behaviour AFAIK. The same warning doesn't appear twice in the same session.

Reply:
Yup, it's default Python behaviour, [can be changed.](https://docs.python.org/3/library/warnings.html)

Reply:
Thanks!